crawl-ref/source/cloud.cc

```diff
@@ -1899,3 +1899,24 @@ void end_still_winds()
     env.level_state &= ~LSTATE_STILL_WINDS;
     mpr("The air resumes its normal movements.");
 }
+
+/**
+ * Surround a monster with clouds of a certain type. This also deletes the same
+ * cloud if it's on top of the monster (which will happen as they walk around).
+ */
+void surround_mons_with_cloud(const monster* mons, cloud_type cloud)
+{
+    const coord_def pos = mons->pos();
+    const cloud_struct* c = cloud_at(pos);
+    if (c && c->type == cloud)
+        delete_cloud(pos);
+    for (adjacent_iterator ai(pos); ai; ++ai)
+    {
+        if (!cell_is_solid(*ai)
+            && (!cloud_at(*ai)
+                || cloud_at(*ai)->type == cloud))
```

Comment:
Skip this check, let cloud rings overwrite their own clouds (so they don't appear to sputter).

Reply:
Doesn't the logic do this? "If (cell can have a cloud and (no cloud OR cloud type matches what we want to place)) { place cloud }"

Reply:
Oh, yes it does.